### ISSUE TITLE
Shown required hidden fields that were hidden by Wordpress' actions.

### ISSUE THREAD

**Issue by martinrios95** [woocommerce/woocommerce#17843]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Add a hook on `functions.php`'s theme file
2. Get that field hidden by type: e.g.: `$fields['shipping']['shipping_city']['type'] = 'hidden';`
3. Update checkout's form page and input some data.

## Expected/actual behavior

When you try to submit form's page, it says that the Shipping City (on this test-case) **is required**, when hidden inputs doesn't require to be filled before submitting the form, because that field **is hidden**. 

## Isolating the problem

- [x ] This bug happens with only WooCommerce plugin active
- [x ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
`
### WordPress Environment ###

Home URL: https://192.168.1.12/wordpress
Site URL: https://192.168.1.12/wordpress
WC Version: 3.1.2
Log Directory Writable: ✔
WP Version: 4.8.3
WP Multisite: –
WP Memory Limit: ❌ 40 MB - We recommend setting memory to at least 64MB. See: Increasing memory allocated to PHP
WP Debug Mode: ✔
WP Cron: ✔
Language: es_AR

### Server Environment ###

Server Info: Apache/2.4.18 (Win32) OpenSSL/1.0.2e PHP/5.6.30
PHP Version: 5.6.30
PHP Post Max Size: 200 MB
PHP Time Limit: 0
PHP Max Input Vars: 1000
cURL Version: 7.51.0
OpenSSL/1.0.2e

SUHOSIN Installed: –
MySQL Version: 5.7.17
Max Upload Size: 200 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.1.2
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (8) ###

Debug Bar: by wordpressdotorg – 0.9
Woo Import Export: by VJInfotech – 2.3
Woocommerce CSV Import: by Allaerd Mensonides – 3.3.5
WooCommerce - Store Exporter: by Visser Labs – 1.9
WooCommerce PayPal Express Checkout Gateway: by WooCommerce – 1.2.1
WooCommerce Table Rate Shipping: by WooCommerce – 3.0.4
WooCommerce: by Automattic – 3.1.2 – 3.2.4 is available
WP Crontrol: by John Blackbourn & contributors – 1.5

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: ARS ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #29 - /shop/
Cart: #30 - /cart/
Checkout: #31 - /checkout/
My account: #32 - /my-account/

### Theme ###

Name: Twenty Seventeen
Version: 1.2 – 1.4 is available
Author URL: https://es-ar.wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by mikejolley** [comment]
Hidden inputs still have values and still exist, so if they are also required **yes this will error.** Remove the input if you don't want it to be validated or posted!

**Comment by mikejolley** [comment]
> Remove the input if you don't want it to be validated or posted

You can also do this via the filters :)

**Comment by martinrios95** [comment]
I'll add a case: if I put a default value, the problem still happens... There's a clue in that case: if I've got a default value, and it's a hidden field, why the form is still requiring a hidden field with a default value?

**Comment by mikejolley** [comment]
How do you know the value was posted/or is populated? Did you check source?

Still if it's not needed, remove it.

**Comment by martinrios95** [comment]
I've used a Wordpress action to insert a default value (e.g: city), and made it hidden. So the field doesn't appear. On my case, it's from shipping and billing to the city the store's come from: Bahia Blanca. Shipping and billing **isn't available on other cities except Bahia Blanca**. But my payment-system doesn't allow reading empty cities or post-codes. So, **I've made it hidden** from shipping-page with a default value, "Bahia Blanca". Simple as that. And there comes this issue.
